packages/block-editor/src/store/private-selectors.js

```diff
@@ -135,3 +135,47 @@ export const isBlockSubtreeDisabled = createSelector(
 	},
 	( state ) => [ state.blockEditingModes, state.blocks.parents ]
 );
+
+/**
+ * Returns a tree of block objects with only clientID and innerBlocks set. Only
+ * blocks with the given editing mode are included in the tree.
+ *
+ * @param {Object}                              state        Global application state.
+ * @param {BlockEditingMode|BlockEditingMode[]} mode         The editing mode(s)
+ *                                                           to filter by.
+ * @param {?string}                             rootClientId Optional root
+ *                                                           client ID of block
+ *                                                           list.
+ *
+ * @return {Object[]} Tree of block objects with only clientID and innerBlocks set.
+ */
+export const getClientIdsTreeWithBlockEditingMode = createSelector(
+	( state, mode, rootClientId = '' ) => {
+		const modes = Array.isArray( mode ) ? mode : [ mode ];
+		return getBlockOrder( state, rootClientId ).flatMap( ( clientId ) => {
+			if ( modes.includes( getBlockEditingMode( state, clientId ) ) ) {
+				return [
+					{
+						clientId,
+						innerBlocks: getClientIdsTreeWithBlockEditingMode(
+							state,
+							mode,
+							clientId
+						),
+					},
+				];
+			}
+			return getClientIdsTreeWithBlockEditingMode(
+				state,
+				mode,
+				clientId
+			);
+		} );
+	},
+	( state ) => [
+		state.blocks.order,
+		state.blockEditingModes,
+		state.settings.templateLock,
+		state.blockListSettings,
+	]
+);
```

Comment:
When a block is disabled and its children aren't, this function will merge the children as the disabled block's siblings. I.e., this:
```
[ { id: 'cover', inner: [ id: 'para1' ], disabled: true }, { id: 'para2' } ]
```
will become:
```
[ { id: 'para1' }, { id: 'para2' } ]
```
This behavior already existed before this PR, but I find it suprising. Do we really want to display the list view tree like this? I would expect such a disabled block to be greyed out.

Reply:
Yeah this was the design decision. It mimics the list view that you get in the right hand sidebar when editing a page or using a content locked pattern. 

We can always revisit based on user feedback 😀